build/bin/user_setup

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+set -x
+
+# ensure $HOME exists and is accessible by group 0 (we don't know what the runtime UID will be)
+mkdir -p ${HOME}
+chown ${USER_UID}:0 ${HOME}
+chmod ug+rwx ${HOME}
```

Comment:
Modifying the user-level permissions here is kind of a no-op, since this happens at build time and user ID is set at runtime. For clarity, I'd suggest we only do a `chgrp -R 0 /home`. 

The full command we use for the arbitrary user ID patch is [here](https://github.com/eclipse/che-devfile-registry/blob/master/arbitrary-users-patch/Dockerfile#L6) and I think it covers all our bases.

Reply:
This file is bootstrapped by the `operator-sdk`, and I'm not sure it would be a good idea to change it.

Reply:
> user ID is set at runtime

The user ID referred to here is the one explicitly defined in the Dockerfile afaict.

Reply:
Do we know of any side effects in case the Dockerfile defines e.g. user 0?

Reply:
Don't know, but Dockerfile and scripts in the `/bin` are generated together by the `operator-sdk` in a way that is expected to be consistent afaict

Reply:
From https://docs.openshift.com/container-platform/3.11/creating_images/guidelines.html#openshift-specific-guidelines
```
RUN chgrp -R 0 /some/directory && \
    chmod -R g=u /some/directory
```
@amisevsk Seems doc suggests to add chmod as well as chgrp. I may don't understand it in details, could you provide complete suggestion we should apply? Or let's do it later as a separate PR.

Reply:
:+1: for separate PR or leaving as-is and trusting the operator-sdk bootstrap.